source/agora/network/NetworkManager.d

```diff
@@ -369,17 +369,24 @@ public class NetworkManager
         However, it can be overriden in test code to return an in-memory client.
 
         Params:
-          address = The address (IPv4, IPv6, hostname) of this node
+          base_url = base part of the REST interface URL
           timeout = the timeout duration to use for requests
 
         Returns:
           An object to communicate with the node at `address`
 
     ***************************************************************************/
 
-    protected API getClient (Address address, Duration timeout)
```

Comment:
Wrong name was used: actually `RestInterfaceClient` accepts base URL string as its first argument

Reply:
`Address` is an alias to `string`

Reply:
Yes. The worst thing is when the name does not describes what is actually stored in the variable. 
I think it is better to prohibit interference of values by using differrent (wrapper) types. It makes things simplier and avoids this problem very well.

Reply:
To clear:
address variable was described as "The address (IPv4, IPv6, hostname) of this node" - this is totally wrong

Reply:
I'm not sure why it would be totally wrong ? The type doesn't enforce it, but it needs to be one of those.

Reply:
Description of `address` was wrong: it contained not "The address (IPv4, IPv6, hostname) of this node", but "base URL". (I not cheched, just use Vibe.d manual of RestInterfaceClient ctor)

My suggestion that all this is because both of them turned to be an alias-masked `string`.

Reply:
The actual type is generic. In unittests we instantiate a RemoteAPI (provided by LocalRest), and the `address` part there must be unique but it can be any string at all. We use it with public keys as the `address`. Have a look here:

https://github.com/bpfkorea/agora/blob/4f6a293ee105c59a49d2af4a88d74a98cf010e5a/source/agora/test/Base.d#L109-L132

Reply:
@AndrejMitrovic hmmm, maybe it is because what `API` describes not "Agora API", but "Agora REST API suitable for using with Vibe.d"

Reply:
(Otherwise, if you try to pass "The address (IPv4, IPv6, hostname) of this node" into implementation of this API called `Node!NetworkManager` it will fail.)